### PR TITLE
feat: remove javascript from exception pages

### PR DESCRIPTION
**Pull request by stakach**
this allows the exceptions to be fully inspected in the preview panes of browsers and postman
making the project much more useful for API apps where requests are driven by XHR

Also updates version setting macro to maintain compatibility with [wyhaines/defined.cr](https://github.com/wyhaines/defined.cr#version-checks)

### CODE REVIEW

**Comment by stakach** [comment]
feel free to edit my branch directly (maintainers can edit)
Uses the tricks found here: https://css-tricks.com/the-checkbox-hack/#aa-tabbed-areas

**Comment by stakach** [comment]
so the page now looks identical for the desktop view, we're working on the mobile view currently

**Comment by stakach** [comment]
okay this is ready for review

**Comment by stakach** [comment]
okay I think I've made a big improvement to inspecting an error on mobile, we're go for merge

File: src/exception_page/version.cr [hunk 1]
```diff
@@ -1,3 +1,7 @@
 class ExceptionPage
-  VERSION = {{ `shards version #{__DIR__}`.chomp.stringify }}
+  macro set_version
+    VERSION = {{ `shards version "#{__DIR__}"`.chomp.stringify.downcase }}
+  end
+
+  set_version
```
Comment (Sija) [inline src/exception_page/version.cr]: This is unrelated change. Also, what's the point of creating such a redundant macro?

Comment (jwoertink) [reply src/exception_page/version.cr]: Yeah, I think we can just keep this as is since it still achieves the goal of not having to manage the version.

Comment (stakach) [reply src/exception_page/version.cr]: maintains compatibility with [wyhaines/defined.cr](https://github.com/wyhaines/defined.cr#version-checks) version check
so you can do stuff like: https://github.com/wyhaines/opentelemetry-instrumentation.cr/blob/main/src/opentelemetry/instrumentation/frameworks/spider-gazelle.cr#L54

**Review by jwoertink — COMMENTED** [review]
Sweet! I just pulled this in to my app, and it looks great on desktop and mobile. I think just the one minor change reverted, and it's all good to go :tada:  Thanks for tackling this!

File: src/exception_page/version.cr (hunk 1 above)
Comment (Sija) [reply src/exception_page/version.cr]: @stakach WDYM by _maintains compatibility_?

Comment (stakach) [reply src/exception_page/version.cr]: basically it makes the VERSION const a little more useful

Comment (Sija) [reply src/exception_page/version.cr]: Also, I don't believe we should have a shard specific-hacks in a general-purpose shard like this.

Comment (Sija) [reply src/exception_page/version.cr]: > basically it makes the VERSION const a little more useful

@stakach I don't see that how that's achieved. Could you explain, please?

Comment (stakach) [reply src/exception_page/version.cr]: it's not about shard specific hacks, it's about being able to look at the VERSION const in macro land
the older implementation is only inspectable at run time - so half as useful

That shard is just an example of why you would implement the const in this way

Comment (jwoertink) [reply src/exception_page/version.cr]: I thought that shard just required the constant to be there? It has to be set from within a macro to work? 

/cc. @wyhaines

Comment (Sija) [reply src/exception_page/version.cr]: You could wrap it in `{% begin %}...{% end %}` then, no need to create a redundant macro.

Comment (Sija) [reply src/exception_page/version.cr]: btw, how come it doesn't work in macro-land? https://carc.in/#/r/d283

Comment (stakach) [reply src/exception_page/version.cr]: good idea, done!

Comment (jwoertink) [reply src/exception_page/version.cr]: oh, that's wild. I just tried it. I see what you mean @stakach 
https://play.crystal-lang.org/#/r/d285

If you take it out of the begin/end, it doesn't print out.

File: src/exception_page/version.cr [hunk 2]
```diff
@@ -1,3 +1,5 @@
 class ExceptionPage
-  VERSION = {{ `shards version #{__DIR__}`.chomp.stringify }}
+  {% begin %}
+    VERSION = {{ `shards version "#{__DIR__}"`.chomp.stringify.downcase }}
```
Comment (Sija) [inline src/exception_page/version.cr]: No need for that.

```suggestion
    VERSION = {{ `shards version "#{__DIR__}"`.chomp.stringify }}
```

File: src/exception_page/version.cr (hunk 1 above)
Comment (stakach) [reply src/exception_page/version.cr]: > I thought that shard just required the constant to be there? It has to be set from within a macro to work?

I found the issue when using @wyhaines shard, so slowly updating everything I've previously written

Comment (jwoertink) [reply src/exception_page/version.cr]: ok, I can dig it.

**Review by jwoertink — APPROVED** [review]
Looks good to me!
